### ISSUE TITLE
Web application for flashcards

### ISSUE THREAD

**Issue by rbeezer** [rbeezer/mathbook#208]
Alex Best wrote a nice app to quiz him on his course notes (written in MBX).  

https://alexjbest.pythonanywhere.com/

Same idea could apply to any MBX textboook.  GPL'ed software at 

https://github.com/alexjbest/blota/

Original discussion at:  https://groups.google.com/d/topic/mathbook-xml-support/f9f4_U6CxlU/discussion

**Comment by alexjbest** [comment]
One issue (in my opinion) with this app, is that the app doesn't know how well you know the cards. It would be ideal for such an app to test you more on the things you get wrong more often, and accept that you know the basics after a while. To take full advantage of https://en.wikipedia.org/wiki/Spaced_repetition. Fortunately I recently found https://fasiha.github.io/ebisu.js/ which if I were to make this again/properly I would definitely use, it's pretty much ideal for this purpose, running in javascript and handling all of the mathematics of recall probability, all that is needed is a nice interface that updates a database of scores for each question (for each user), along with the code that extracts good questions from a PTX source.

**Comment by kcrisman** [comment]
There are a number of these spaced repetition apps.  I wonder if any ideas from https://github.com/dae/anki would be useful here.

**Comment by alexjbest** [comment]
@kcrisman Yes thats a great point, anki has spaced repetition as per <https://apps.ankiweb.net/docs/manual.html#what-spaced-repetition-algorithm-does-anki-use> (implemented in https://github.com/dae/anki/blob/master/anki/sched.py), and some built in Latex support for math (and there are other methods too <https://niklaskorz.de/2017/06/studying-mathematics-with-anki-and-mathjax.html>).

Perhaps that means anki or other general flashcard apps are the way to go and writing an Anki-exporter rather than a new webapp is the best use of time.

The only disadvantage I see is for instructors, who may wish to see metrics on students understanding, this approach would likely involve relinquishing some control there.

**Comment by kcrisman** [comment]
Well, you can do what makes the most sense for you; I was only noting that Anki is AGPL and already does this.  I like it a lot myself.  Yes, exporting Anki cards as a new stylesheet would be a good first exercise.

You may also want to get in touch with the folks at Ximera (https://github.com/XimeraProject) such as @kisonecat who would be very interested in the metrics side of things as well as the spacing.

**Comment by kisonecat** [comment]
I also use and like Anki.  It's helped me study for exams.

With Ximera, we've certainly tried a lot in terms of metrics for assessment and interleaved practice.  We've done IRT on the multiple-choice questions asked within Ximera, for instance.

[Are you wanting to use something like this in a graded face-to-face course?  When considering more complicated course-wide assessment schemes (say doing PCA on the gradebook), there are challenges not only on the instructor-facing side but also on the student-facing side.  Students need to know how much time the homework will take them, because they're managing other courses as well -- they're trying to understand the marginal return on grade for so much studying to solve their personal constrained optimization problem.  So systems that are producing more accurate insights about student learning may end up being pretty confusing for students.  This bleeds over to how something like ALEKS gets used in courses too, e.g., for homework, you see some instructors assign so many minutes of engagement with ALEKS, rather than relying on the metrics that ALEKS produces.]
